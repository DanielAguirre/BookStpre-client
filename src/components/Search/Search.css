.search {
  width: 100%;
  height: 20px;
  margin-bottom: 40px;
  display: flex;
  flex-direction: row;
}

.search-input {
  width: 30%;
  margin-right: 15px;
}

.search-query {
  /*background-image: url('./');*/
}