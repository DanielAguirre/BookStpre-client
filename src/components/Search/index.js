import React from 'react';
import glamorous from 'glamorous';
import { Input } from '../Form';

const Div = glamorous.div({
  width: '100%',
  height: '20px',
  marginBottom: '40px',
  display: 'flex',
  flexDirection: 'row'
});

const InputSearch = glamorous(Input)({
  width: '30%',
  marginRight: '15px',
});

const Search = ({ foundBooks, titleValue, authorValue, priceValue, handleInputChange }) => (
  <Div className="search">
    <InputSearch 
      id="searchTitle"
      name="isGoing"
      className="search-input"
      placeholder="Title"
      onChange={handleInputChange}
      value={titleValue}/>
    <InputSearch  
      id="searchAuthor"
      name='authorInput'
      className="search-input" 
      placeholder="Author"
      onChange={handleInputChange}
      value={authorValue} />
    <InputSearch  
      id="searchPrice"
      name='priceInput'
      className="search-input" 
      placeholder="Price" 
      onChange={handleInputChange}
      value={priceValue} />
    <button id="search" className="search-query" onClick={foundBooks}> Search </button>
  </Div>
);

module.exports = Search