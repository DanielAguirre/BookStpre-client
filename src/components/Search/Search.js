import React from 'react';
import './Search.css';

const Search = () => (
  <div className="search">
    <input id="searhTitle" className="search-input" placeholder="Title" />
    <input id="searchAuthor" className="search-input" placeholder="Author"/>
    <input id="searchPrice" className="search-input" placeholder="Price"/>
    <button id="search" className="search-query"> Search </button>
  </div>
);

module.exports = Search