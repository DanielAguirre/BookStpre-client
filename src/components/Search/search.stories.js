import React from 'react';
import { storiesOf, action, linkTo } from '@kadira/storybook';
import Search from './index';

storiesOf('Search', module)
  .add('Search fields', () => (
    <Search />
  ))

