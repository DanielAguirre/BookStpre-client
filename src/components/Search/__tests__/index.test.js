import React from 'react';
import { shallow, mount } from 'enzyme';
import Search from '../index';

it('<Search />, Should reder the searrh component', () => {
  function foundBooks(e) {
    expect(true).toBeTruthy();
  };

  const titleInputRef = (input) => {
    expect(input).toBeTruthy();
  };

  const searchWrapper = shallow(<Search 
    titleInputRef={titleInputRef}
    foundBooks={foundBooks}/>
  );
  // Render component
  expect(searchWrapper).toMatchSnapshot();
  
  // test props
  expect(searchWrapper.instance().props.titleInputRef).toEqual(titleInputRef);
  expect(searchWrapper.instance().props.foundBooks).toEqual(foundBooks);
  searchWrapper.find('#search').simulate('click')
})
