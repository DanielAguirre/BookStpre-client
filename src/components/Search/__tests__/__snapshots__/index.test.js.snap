exports[`test <Search />, Should reder the searrh component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <glamorous(div)
    className="search">
    <glamorous(unknown)
        className="search-input"
        id="searchTitle"
        name="isGoing"
        onChange={undefined}
        placeholder="Title"
        value={undefined} />
    <glamorous(unknown)
        className="search-input"
        id="searchAuthor"
        name="authorInput"
        onChange={undefined}
        placeholder="Author"
        value={undefined} />
    <glamorous(unknown)
        className="search-input"
        id="searchPrice"
        name="priceInput"
        onChange={undefined}
        placeholder="Price"
        value={undefined} />
    <button
        className="search-query"
        id="search"
        onClick={[Function]}>
         Search 
    </button>
</glamorous(div)>,
  "nodes": Array [
    <glamorous(div)
      className="search">
      <glamorous(unknown)
            className="search-input"
            id="searchTitle"
            name="isGoing"
            onChange={undefined}
            placeholder="Title"
            value={undefined} />
      <glamorous(unknown)
            className="search-input"
            id="searchAuthor"
            name="authorInput"
            onChange={undefined}
            placeholder="Author"
            value={undefined} />
      <glamorous(unknown)
            className="search-input"
            id="searchPrice"
            name="priceInput"
            onChange={undefined}
            placeholder="Price"
            value={undefined} />
      <button
            className="search-query"
            id="search"
            onClick={[Function]}>
             Search 
      </button>
</glamorous(div)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Search
        foundBooks={[Function]}
        titleInputRef={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "foundBooks": [Function],
          "titleInputRef": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <glamorous(div)
          className="search">
          <glamorous(unknown)
                    className="search-input"
                    id="searchTitle"
                    name="isGoing"
                    onChange={undefined}
                    placeholder="Title"
                    value={undefined} />
          <glamorous(unknown)
                    className="search-input"
                    id="searchAuthor"
                    name="authorInput"
                    onChange={undefined}
                    placeholder="Author"
                    value={undefined} />
          <glamorous(unknown)
                    className="search-input"
                    id="searchPrice"
                    name="priceInput"
                    onChange={undefined}
                    placeholder="Price"
                    value={undefined} />
          <button
                    className="search-query"
                    id="search"
                    onClick={[Function]}>
                     Search 
          </button>
</glamorous(div)>,
        "_debugID": 2,
        "_renderedOutput": <glamorous(div)
          className="search">
          <glamorous(unknown)
                    className="search-input"
                    id="searchTitle"
                    name="isGoing"
                    onChange={undefined}
                    placeholder="Title"
                    value={undefined} />
          <glamorous(unknown)
                    className="search-input"
                    id="searchAuthor"
                    name="authorInput"
                    onChange={undefined}
                    placeholder="Author"
                    value={undefined} />
          <glamorous(unknown)
                    className="search-input"
                    id="searchPrice"
                    name="priceInput"
                    onChange={undefined}
                    placeholder="Price"
                    value={undefined} />
          <button
                    className="search-query"
                    id="search"
                    onClick={[Function]}>
                     Search 
          </button>
</glamorous(div)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Search
    foundBooks={[Function]}
    titleInputRef={[Function]} />,
}
`;
